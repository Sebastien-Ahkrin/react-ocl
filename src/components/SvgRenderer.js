import React from 'react';
import propTypes from 'prop-types';

export default function SvgRenderer(props) {
  const html = {
    __html: props.mol.toSVG(props.width, props.height, props.id, props)
  };

  // eslint-disable-next-line react/no-danger
  return <div dangerouslySetInnerHTML={html} />;
}

SvgRenderer.propTypes = {
  width: propTypes.number,
  height: propTypes.number,
  id: propTypes.string,
  factorTextSize: propTypes.number,
  fontWeight: propTypes.string,
  strokeWidth: propTypes.string,
  inflateToMaxAVBL: propTypes.bool,
  inflateToHighResAVBL: propTypes.bool,
  chiralTextBelowMolecule: propTypes.bool,
  chiralTextAboveMolecule: propTypes.bool,
  chiralTextOnFrameTop: propTypes.bool,
  chiralTextOnFrameBottom: propTypes.bool,
  noTabus: propTypes.bool,
  showAtomNumber: propTypes.bool,
  showBondNumber: propTypes.bool,
  highlightQueryFeatures: propTypes.bool,
  showMapping: propTypes.bool,
  suppressChiralText: propTypes.bool,
  suppressCIPParity: propTypes.bool,
  suppressESR: propTypes.bool,
  showSymmetrySimple: propTypes.bool,
  showSymmetryDiastereotopic: propTypes.bool,
  showSymmetryEnantiotopic: propTypes.bool,
  noImplicitAtomLabelColors: propTypes.bool,
  noStereoProblem: propTypes.bool
};

SvgRenderer.defaultProps = {
  width: 300,
  height: 150,
  suppressChiralText: true,
  suppressESR: true,
  suppressCIPParity: true,
  noStereoProblem: true,
  factorTextSize: false,
  fontWeight: false,
  strokeWidth: false,
  inflateToMaxAVBL: false,
  inflateToHighResAVBL: false,
  chiralTextBelowMolecule: false,
  chiralTextAboveMolecule: false,
  chiralTextOnFrameTop: false,
  chiralTextOnFrameBottom: false,
  noTabus: false,
  showAtomNumber: false,
  showBondNumber: false,
  highlightQueryFeatures: false,
  showMapping: false,
  showSymmetrySimple: false,
  showSymmetryDiastereotopic: false,
  showSymmetryEnantiotopic: false,
  noImplicitAtomLabelColors: false
};
