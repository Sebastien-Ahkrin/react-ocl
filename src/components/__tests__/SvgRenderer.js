import React from 'react';
import renderer from 'react-test-renderer';

import SvgRenderer from '../SvgRenderer';

test('Molecule renders', () => {
    const component = renderer.create(
        <SvgRenderer smiles="CCOC" width={1200} height={800}/>
    );
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
