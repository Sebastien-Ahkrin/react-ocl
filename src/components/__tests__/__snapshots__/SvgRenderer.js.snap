exports[`test Molecule renders 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg id=\"mol1\" xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\" width=\"1200px\" height=\"800px\" viewBox=\"0 0 1200 800\">
    	<style> #mol1 {pointer-events:none; }  #mol1 .event  { pointer-events:all;}  </style>
    	<text x=\"584\" y=\"398\" font-family=\" Helvetica\" font-size=\"14\" fill=\"rgb(255,13,13)\">O</text>
    	<line x1=\"610\" y1=\"406\" x2=\"631\" y2=\"394\" style=\"stroke:rgb(0,0,0);stroke-width:1\"/>
    	<line x1=\"596\" y1=\"397\" x2=\"610\" y2=\"406\" style=\"stroke:rgb(0,0,0);stroke-width:1\"/>
    	<line x1=\"568\" y1=\"406\" x2=\"583\" y2=\"397\" style=\"stroke:rgb(0,0,0);stroke-width:1\"/>
    	<line id=\"mol1:Bond:1-0\" class=\"event\" x1=\"610\" y1=\"406\" x2=\"631\" y2=\"394\" stroke-width=\"8\" stroke-opacity=\"0\"/>
    	<line id=\"mol1:Bond:2-1\" class=\"event\" x1=\"589\" y1=\"394\" x2=\"610\" y2=\"406\" stroke-width=\"8\" stroke-opacity=\"0\"/>
    	<line id=\"mol1:Bond:3-2\" class=\"event\" x1=\"568\" y1=\"406\" x2=\"589\" y2=\"394\" stroke-width=\"8\" stroke-opacity=\"0\"/>
    	<circle id=\"mol1:Atom:0\" class=\"event\" cx=\"631\" cy=\"394\" r=\"8\" fill-opacity=\"0\"/>
    	<circle id=\"mol1:Atom:1\" class=\"event\" cx=\"610\" cy=\"406\" r=\"8\" fill-opacity=\"0\"/>
    	<circle id=\"mol1:Atom:2\" class=\"event\" cx=\"589\" cy=\"394\" r=\"8\" fill-opacity=\"0\"/>
    	<circle id=\"mol1:Atom:3\" class=\"event\" cx=\"568\" cy=\"406\" r=\"8\" fill-opacity=\"0\"/>
    </svg>",
    }
  } />
`;
